src/query/api/v1/handler/fetch_options.go

```diff
@@ -105,3 +136,81 @@ func newOrExistingRestrictFetchOptions(
 	}
 	return &storage.RestrictFetchOptions{}
 }
+
+// ParseStep parses the step duration for an HTTP request.
+func ParseStep(r *http.Request) (time.Duration, bool, error) {
+	step := r.FormValue(StepParam)
+	if step == "" {
+		return 0, false, nil
+	}
+	value, err := parseStep(r, StepParam)
+	if err != nil {
+		return 0, false, err
+	}
+	return value, true, err
+}
+
+func parseStep(r *http.Request, key string) (time.Duration, error) {
+	step, err := ParseDuration(r, key)
+	if err != nil {
+		return 0, err
+	}
+	if step <= 0 {
+		return 0, fmt.Errorf("expected postive step size, instead got: %d", step)
+	}
+	return step, nil
+}
+
+// ParseLookbackDuration parses a lookback duration for an HTTP request.
+func ParseLookbackDuration(r *http.Request) (time.Duration, bool, error) {
+	lookback := r.FormValue(LookbackParam)
+	if lookback == "" {
+		return 0, false, nil
+	}
+
+	if lookback == StepParam {
+		// Use the step size as lookback.
+		step, err := parseStep(r, StepParam)
+		if err != nil {
+			return 0, false, err
+		}
+		return step, true, nil
+	}
+
+	// Otherwise it is specified as duration value.
+	value, err := parseStep(r, LookbackParam)
+	if err != nil {
+		return 0, false, err
+	}
+
+	return value, true, nil
+}
+
+// ParseDuration parses a duration HTTP parameter.
+// nolint: unparam
+func ParseDuration(r *http.Request, key string) (time.Duration, error) {
+	str := strings.TrimSpace(r.FormValue(key))
+	if str == "" {
+		return 0, errors.ErrNotFound
+	}
+
+	value, durationErr := time.ParseDuration(str)
+	if durationErr == nil {
+		return value, nil
+	}
+
+	// Try parsing as a float value specifying seconds, the Prometheus default
+	seconds, floatErr := strconv.ParseFloat(str, 64)
+	if floatErr == nil {
+		ts := seconds * float64(time.Second)
+		if ts > maxInt64 || ts < minInt64 {
```

Comment:
Should the lower bound just check `ts <= 0`?

Reply:
This was existing code, I could update though.